packages/framework/src/Model/Product/ProductPromotionXy.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Product;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(name="product_promotion_xy")
+ * @ORM\Entity
+ */
+class ProductPromotionXy
+{
+    /**
+     * @var int
+     * @ORM\Column(type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="IDENTITY")
+     */
+    protected $id;
+
+    /**
+     * @var int
+     * @ORM\Column(type="integer", nullable=false, name="buy_quantity")
+     */
+    protected $buyQuantity;
+
+    /**
+     * @var int
+     * @ORM\Column(type="integer", nullable=false, name="free_quantity")
+     */
+    protected $freeQuantity;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductPromotionXyData $productPromotionXyData
+     */
+    public function __construct(ProductPromotionXyData $productPromotionXyData)
+    {
+        $this->setData($productPromotionXyData);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductPromotionXyData $productPromotionXyData
+     * @return \Shopsys\FrameworkBundle\Model\Product\ProductPromotionXy
+     */
+    public static function create(ProductPromotionXyData $productPromotionXyData)
+    {
+        return new static($productPromotionXyData);
```

Comment:
I'm not quite sure about this method. I think we should use only constructor and factory methods to create objects. Not the static create method